src/utils/transactions.ts

```diff
@@ -84,3 +84,86 @@ export const useTransactionInfo = (tx: Transaction | MempoolTransaction, address
     lockTime
   }
 }
+
+// TODO: The following 2 functions could be ported to js-sdk (and properly tested there)
+type AttoAlphAmount = string
+type Address = string
+
+type UTXO = {
+  attoAlphAmount?: AttoAlphAmount
+  address?: Address
+  tokens?: { id: string; amount: string }[]
+}
+
+export const sumUpAlphAmounts = (utxos: UTXO[]): Record<Address, AttoAlphAmount> => {
+  const validUtxos = utxos.filter((utxo) => utxo.address && utxo.attoAlphAmount)
+
+  const grouped = groupBy(validUtxos, 'address')
+  const summed = mapValues(grouped, (addressGroup) =>
+    reduce(addressGroup, (sum, utxo) => (BigInt(sum) + BigInt(utxo.attoAlphAmount || 0)).toString(), '0')
+  )
+
+  return summed
+}
+
+export const sumUpTokenAmounts = (utxos: UTXO[]): Record<Address, Record<string, string>> => {
+  const validUtxos = utxos.filter((utxo) => utxo.address && utxo.tokens && utxo.tokens.length > 0)
+
+  const grouped = groupBy(validUtxos, 'address')
+  const summed = mapValues(grouped, (addressGroup) => {
+    const tokenSums: Record<string, string> = {}
+    for (const utxo of addressGroup) {
+      for (const token of utxo.tokens || []) {
+        if (!tokenSums[token.id]) {
+          tokenSums[token.id]
+        }
+        tokenSums[token.id] = (BigInt(tokenSums[token.id] || 0) + BigInt(token.amount)).toString()
+      }
```

Comment:
While still trying to understand the algorithm I wonder if this implementation would be easier to grasp:

```ts
tokenSums[token.id] = !tokensSums[token.id] ? token.amount : (BigInt(tokenSums[token.id]) + BigInt(token.amount)).toString()
```


```

Reply:
As said before, I think the condition is obsolete. This safe guard was needed at one point during my iterations, but based on the current logic and my recent testing it works well without it. To be tested further.